compile/compile.go

```diff
@@ -483,8 +486,28 @@ func (c *Compiler) compilePlan(ctx context.Context) error {
 
 	// Prepare modules and builtins for the planner.
 	modules := []*ast.Module{}
+	allPaths := ast.NewSet()
 	for _, module := range c.compiler.Modules {
 		modules = append(modules, module)
+
+		modulePath, err := module.Package.Path.Ptr()
+		if err != nil {
+			return err
+		}
+		allPaths.Add(ast.StringTerm(modulePath))
+
+		for _, rule := range module.Rules {
+			rulePath, err := rule.Path().Ptr()
+			if err != nil {
+				return err
+			}
+			allPaths.Add(ast.StringTerm(rulePath))
+		}
+	}
```

Comment:
These changes can be removed if we rely on the rule tree by calling GetRulesExact (or GetRules) above.